sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -1425,19 +1390,43 @@ private List<Long> getIndexListIds(PersonCriteria criteria, Integer first, Integ
 		final PersonJoins personJoins = personQueryContext.getJoins();
 		personJoins.configure(criteria);
 
-		List<Selection<?>> selects = new ArrayList<>();
-		selects.add(person.get(Person.ID));
+		List<Selection<?>> selections = new ArrayList<>();
+		selections.add(person.get(Person.ID));
+		selections.addAll(addSortProperties(sortProperties, personQueryContext));
+
+		cq.multiselect(selections);
+
+		Predicate filter = createIndexListFilter(criteria, personQueryContext);
+		if (filter != null) {
+			cq.where(filter);
+		}
+		cq.distinct(true);
+
+		List<Tuple> persons = QueryHelper.getResultList(em, cq, first, max);
 
-		if (sortProperties != null && sortProperties.size() > 0) {
-			List<Order> order = new ArrayList<Order>(sortProperties.size());
+		logger.trace(
+			"getIndexListIds() finished. association={}, count={}, {}ms",
+			Optional.ofNullable(criteria).orElse(new PersonCriteria()).getPersonAssociation().name(),
+			persons.size(),
+			DateHelper.durationMillies(startTime));
+		return persons.stream().map(t -> t.get(0, Long.class)).collect(Collectors.toList());
+	}
+
+	private List<Selection<?>> addSortProperties(List<SortProperty> sortProperties, PersonQueryContext personQueryContext) {
```

Comment:
Please rename to `sortBy` to point out that the logic does not only compile the sort selections, but also applies them.

Reply:
Done.